### DISCUSSION TITLE
Photo Collection Folders (send-only) - test alpha release on TestFlight

### DISCUSSION THREAD

**Discussion by SimonPickup** [MobiusSync/MobiusSync#172] [Beta]
The first **alpha** release of Möbius Sync supporting _sendonly_ **Photo Collection** folders (originally #4) is now available via TestFlight.

Please subscribe to this discussion for updates and **please report your experiences here**.

The initial feature set is quite limited:
- Choose any photo collection (e.g. Recents, Favourites, Albums, Smart Albums)
- Will sync up to 5000 photos/videos in a _sendonly_ mode, i.e. send one way to a syncthing peer
- Includes all photos/videos, excluding hidden items
- Files will be named in the form 2024-06/2024-06-29T200657_IMG_7231.HEIC (in future could be configurable)
- Filename conflicts are not yet handled
- Multiple folders are supported
- No special support for Live Photos or Bursts as yet
- Not exactly sure how edits are supported
- "Original" file content is synced, i.e. no manipulation is done by Möbius Sync
- Background sync is working in the same way as for other files
- Incremental photo collection changes are handled more efficiently on recent iOS versions
- Haven't tested versioning

Future ideas (in no particular order) are:
- Configurable whether to include photos and/or videos, filtering by image type/size
- Configurable whether to include hidden items
- Configurable max number of items or possibly date ranges
- Configurable output pathname template
- How to handle pathname collisions
- Support for image/vide edits and versioning
- Support for different "output" forms/variants for Live Photos, Bursts, stills from videos etc
- Would be nice to be able to do optional metadata stripping (e.g. geolocation) but will likely be challenging to do this without a large performance hit
- Creation/modification/capture date handling/differentiation
- Obviously bidirectional sync would be lovely but will definitely be complicated so we plan to refine sendonly first

For Möbius Sync (Freemium), register here: https://testflight.apple.com/join/sBZ8ZuCV
For Möbius Sync Pro, register here: https://testflight.apple.com/join/FokR8tfx

The above TestFlight groups are distinct to previous Möbius Sync beta testing groups - please sign up again. Membership is limited to 100 users initially.

### Build 202406301840 is now available, including:
#4 (minimal functionality as described above)
Bug fixes:
#140
#167 
Planned:
#170

**Comment by tlaurion** [comment]
@SimonPickup what are the blockers of permitting deletion of files that would have been synced but moved from the destination of the sync (to remove them from origin, to liberate space and reduce costs of cloud).

One usage of synthing on all other platforms but Apple today is the possibility to free space on lower storage decides and stop relying on paid cloud services.

To be honest, with this project still not existing, it is really clear to Apple users that they must also agree on paying cloud services. This project could change that, but not quite there yet.

Agree this is first step, having one direction sync is one important step. Two ways sync would be amazing, but I think a middle step is missing here. And that middle step is permit to remove files from Apple devices that have been synced, but then moved/removed from non-apple destination (clearly stating that the file is not needed on apple origin, to free space, as well as on icloud to reduce cloud payments as well to eventually none.).

**Reply by SimonPickup** [reply]
I have split this off to a new feature request #173.

**Comment by SimonPickup** [comment]
Which of the future ideas listed above do you need to make this usable?

**Comment by SimonPickup** [comment]
TestFlight build 202407030520 is now released, merging in Syncthing 1.27.9. I have also now shared on the previous TestFlight beta groups.

**Comment by SimonPickup** [comment]
TestFlight build 202407060900 is now released. The limit of 5000 photos/videos per collection has been lifted to 100,000, which is intended to be a safeguard, assuming nobody will have more than 100,000. In future this will be configurable anyway.

**Reply by NightMean** [reply]
Thanks for the updates! Do you have anything new in regards of the live photos? That's what I'm mostly interested about currently.

**Reply by SimonPickup** [reply]
> Thanks for the updates! Do you have anything new in regards of the live photos? That's what I'm mostly interested about currently. 

Live Photo videos will likely be coming next. Great to receive your feedback, thanks.

What does everyone else want next?

**Comment by SimonPickup** [comment]
Testflight build 202407201420 now released:
1. Added support for Live Photos to be sent as photos and/or videos
2. Added Photo Collection tab in Folder Options to toggle on/off photos and videos
3. Added "All Photos" option (effectictively equivalent to Recents) when choosing a photo collection.
4. Photo filename conflicts will be disambiguated, e.g. duplicate images.

**Reply by andreasnyc** [reply]
@SimonPickup - thanks for all the work on syncing photos! The most significant missing feature to me is to have Live Photos synced intact in one file with the still image and Live Photo video all in one image file. This is what happens if I do a google photos backup of the Live Photo. I think this is not what happens currently. This is a blocker for me as my use case is:

iPhone (MobisusSync) --sync all photos & videos--> Google Pixel 1( syncthing)
google pixel 1 (google photos android app) --backup all photos & videos--> Google Photos cloud

The desired behavior is that photos and videos get to Google Photos cloud the same as they would if backed up directly on the iPhone (i.e. Live Photos remain a single file containing the photo & short video).

**Reply by NightMean** [reply]
iOS never stored the live photo in one single file. It is always 2 separate files. JPG/HEIC along with a MOV that is combined using the file metadata that iOS writes. If you download iOS Live Photo from Google Photos, you'll always have 2 separate files. 

If you want to have it as a single JPG file, you can convert it to Google Motion Photo using https://github.com/mihir-io/MotionPhotoMuxer
For HEIC file, it's more of a hack than official support but single HEIC can also be in the format of Google/Samsung Motion Photo if you convert it by this tool https://github.com/PetrVys/MotionPhoto2

**Reply by andreasnyc** [reply]
@NightMean - I want it to end-up in Google Photos just as it would be if I directly backed it up from the iPhone using Google Photos. When that happens, in Google Photos it is a single photo-with-live-photo instead of a photo a video a photo a video which breaks the paradigm of "a photo with live video". Google Photos does this fine while it backs-up the live photos so it can be done integrally to the transfer so that's what I'm asking if can happen. I think others would benefit vs. having to use synching via MobiusSync then an add-on process just so they don't end-up with 2x the files and a whole bunch of videos that aren't really 'videos' taken as such, but live photo 2 sec videos. :)

**Reply by SimonPickup** [reply]
Thanks @NightMean for your explanation and suggested workarounds. Yes, iOS stores and exposes the still photo and video components as two separate filesa, and presumably Google Photos does too (since that's how you get them when downloaded), and presumably the iOS Google Photos app stores the association between the two so they can be displayed in Google Photos as one entity. Exporting via Android (using MöbiusSync or any other method) presumably breaks that association. I'm not sure there's much I can do about that.

@andreasnyc, in case it gives me any inspiration, can you explain why you want to sync via Android first, rather than just running the iOS Google Photos app independently on your iPhone and your Google Pixel?

**Reply by jdarpinian** [reply]
Photos backed up from Google Photos running on a Google Pixel 1 device get free cloud storage in original quality, unlimited, forever. Even photos synced from other devices. That's my use case for this feature and I suspect a lot of other people as well.

**Reply by SimonPickup** [reply]
Ah, I assumed it was a per-google-account setting, not restricted to the device uploading. Now I understand, thanks!

**Reply by SimonPickup** [reply]
I have created a new issue for this: #199. Let's continue the discussion there.
